compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2074,6 +2078,10 @@ object Types {
     }
   }
 
+  trait Softenable {
```

Comment:
Do we need a trait for that? Or could `isSoft` be a method of `Type` directly?

To-do:
- [ ] Find a better name
- [ ] Documentation

Reply:
It's never consumed as a this upper type, and there's no "shared implementation".  So I'd just define/redefine it on SingletonType/ConstantType, and maybe like the docs if they're extensive.  Just IMO :smile: